src/Aspire.Hosting.Azure/ProvisioningExtensions.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Publishing;
+using Microsoft.Extensions.Configuration;
+
+namespace Aspire.Hosting.Azure;
+internal static class ProvisioningExtensions
+{
+    internal static void WriteBicepResourceToManifest(ManifestPublishingContext context, IResource resource)
+    {
+        context.Writer.WriteString("type", "azure.bicep.v0");
+
+        // Grab environment variables that would be passed in from AZD.
```

Comment:
@tg-msft @JoshLove-msft @m-nash I'm still trying to understand the whole IConstruct mechanism but I'm not sure why I need a tenant ID and subscription ID when I am just generating the Bicep?

Reply:
Without these we cannot know the full context of the resource which means we will have a much more difficult time constructing all the correct dependencies.

I think we can abstract out the Tenant as we don't support multi-tenant and can simply use the tenant() function in bicep, but with multi-subscription we would need to know what subscription is being used.

Right now my understanding is that azd requires you to have an env variable AZURE_SUBSCRIPTION_ID otherwise you get promoted for the subscription when you deploy.  We are just using the same thing at an earlier step here.

Reply:
Currently AZD invokes the AppHost before the subscription is known when the `azd init` command is executed. The `azd init` command needs to the manifest (including Bicep files) to help configure which services will be exposed.

I'm still not sure why we need the subscription ID (or even the resource group) to be known. Why can't we just emit resource blocks without it having to be scoped to a resource group?

Reply:
Agreed

Reply:
We can create overloads that allow you to initialize with "unknown sub / tenant" and Aspire can call these overloads.  We can then add the params to the bicep so azd can pass those in either from user prompt or their standard env variables they use.